### PR TITLE
Use-after-frees

### PR DESCRIPTION
**Pull request by ailin-nemui**
coypu AT sdf DOT org wrote:

> Hi folks!
> 
> I tried clang-static-analyzer on irssi 1.1.1, it seems like it
> finds some things. Here's a diff, but there might be more that you would
> want to check, or choose to work differently.
> 
> (in special-vars.c, ret is commands->data sometime)
> 
> I hope it's not too much trouble if reported as a confidential bug.
> Thanks.

### CODE REVIEW

File: src/core/modules.c [hunk 1]
```diff
@@ -289,8 +289,9 @@ void modules_deinit(void)
 
 	while (list != NULL) {
 		module_uniq_destroy(list->data);
-		g_free(list->data);
+		gconstpointer tmp = list->data;
 		list = g_slist_remove(list, list->data);
+		g_free(tmp);
```
Comment (LemonBoy) [inline src/core/modules.c]: `g_free` doesn't like the `gconstpointer`

Comment (ailin-nemui) [reply src/core/modules.c]: it's a bug in the C language

Comment (LemonBoy) [reply src/core/modules.c]: Not the only one and not the worst one but still :)

**Comment by dequis** [comment]
Turns out the dcc part of the patch changed a use after free from "referencing the address of an object which just got deallocated without touching its contents" to "reading an address out of an object that just got deallocated" making it crash on exist. Cool. That's fixed now.

**Comment by ailin-nemui** [comment]
cool

**Comment by ailin-nemui** [comment]
@irssi/developers

**Pull request merged**